### PR TITLE
Add tests and fix bugs on ApplicationLogStream

### PR DESCRIPTION
**Pull request by hsablonniere**
This PR introduces a test suite for ApplicationLogStream along with a series of bugfixes for the retry mechanism and error handling. It also adds more explicit errors.

For now, it's tested on Node.js but with some additional work it could be run automatically in a browser.

## Errors

* `NetworkError`
  * DNS propblem
  * server stops or is not listening
  * TCP connection closes
  * Retry max reached
  * Timeout (no events, heartbeat or logs)
* `HttpError` with `status` property
  * status 4xx
  * status 5xx
* `ServerError`
  * status 200 + bad `content-type`
  * server response closes without a clean ended by

## Fixes bugs and behaviors

* The heartbeat `setInterval` was not cleared in many situations
  * when `.close()` is called
  * when a timeout error occurs
* Most `start.then()` have a `reason` set to an Error object with `DOMException [AbortError]: This operation was aborted`
* When the server stopped or closed the response, the rejected "error" is a `string` and not an `Error`
  * See `this.reject(err + ': ' + error);`
  * The text is: `retry is not enabled: XXXXXXX`
  * `retry is not enabled` should not be an error
  * For responses closed by the server, we had a bug in sse-parse `getBytes`
    * `Cannot read properties of undefined (reading 'length')`
* When a timeout (no heartbeat) error occured:
  * an error was emitted `no healthcheck since 4899, restarting...`
    * but it was emitted after the retry wait period and not directly
  * an inner call to `pause` and then `resume` was done just after an actually triggering a new fetch
    * this makes a really weird behaviour when retry is enabled as a retry is done without any delay
* Without any logs or heartbeats, the healthcheck woud never fail
  * `lastContact` is only updated in onMessage
  * and `performHealthCheck` ignores `lastContact == null`
* Apart from 401, all HTTP errors allowed a retry to be done
  * 400 and 404 should not retry
* When the content type was not SSE or JSON, the error was about the content type
  * I think a 500 with text/plain should error for the status code before the content type
* The max retry was reached in a weird way (hard to explain)
* The SSE event IDs were not retrieve and thus
  * The header `last-event-id` was empty on pause/resume and error/retry
  * The limit query parameter was not updated on pause/resume and error/retry

### CODE REVIEW

File: esm/streams/clever-cloud-sse.js [hunk 1]
```diff
@@ -289,5 +319,15 @@ function formatValue (value) {
   return value.toString();
 }
 
-export class AuthenticationError extends Error {
-};
```
Comment (miton18) [inline esm/streams/clever-cloud-sse.js:293]: Do we loose AuthenticationError ?

Comment (hsablonniere) [reply esm/streams/clever-cloud-sse.js:293]: We have NetworkError with status now.

**Comment by miton18** [comment]
QA:

```
2024-01-22T17:04:38.519Z: time="2024-01-22T17:04:38Z" level=info msg="Get '-'"
2024-01-22T17:04:38.522Z: time="2024-01-22T17:04:38Z" level=info msg="Get '-'"
2024-01-22T17:04:38.897Z: time="2024-01-22T17:04:38Z" level=info msg="Get '-'" <- cut internet
[STACKTRACE]
NetworkError: Failed to receive heartbeat within 4000ms period
    at Timeout.<anonymous> (/home/miton18/code/nodejs/clever-client.js/cjs/streams/clever-cloud-sse.js:227:23)
    at listOnTimeout (node:internal/timers:569:17)
    at process.processTimers (node:internal/timers:512:7)
[/STACKTRACE]
[ERROR] Failed to receive heartbeat within 4000ms period

**Comment by miton18** [comment]
No retry 
```
2024-01-29T15:28:24.827Z: time="2024-01-29T15:28:24Z" level=info msg="Get '-'"
2024-01-29T15:28:25.158Z: time="2024-01-29T15:28:25Z" level=info msg="Get '-'"
[STACKTRACE]
NetworkError: Failed to receive heartbeat within 4000ms period
    at Timeout.<anonymous> (/home/miton18/code/nodejs/clever-client.js/cjs/streams/clever-cloud-sse.js:227:23)
    at listOnTimeout (node:internal/timers:569:17)
    at process.processTimers (node:internal/timers:512:7)
[/STACKTRACE]
[ERROR] Failed to receive heartbeat within 4000ms period

**Comment by miton18** [comment]
with retry
```
2024-01-29T15:30:29.986Z: time="2024-01-29T15:30:29Z" level=info msg="Get '-'"
2024-01-29T15:30:29.986Z: time="2024-01-29T15:30:29Z" level=info msg="Get '-'"
2024-01-29T15:30:30.047Z: time="2024-01-29T15:30:30Z" level=info msg="Get '-'"
2024-01-29T15:30:30.273Z: time="2024-01-29T15:30:30Z" level=info msg="Get '-'"
[STACKTRACE]
an error occured: undefined
[/STACKTRACE]
[ERROR] an error occured: undefined
[DEBUG] stream opened! {"appId":"app_fcdbc590-c6d1-41c1-bc22-7b06d0ea76e5","filter":null,"deploymentId":null}
2024-01-29T15:30:30.861Z: time="2024-01-29T15:30:30Z" level=info msg="Get '-'"
2024-01-29T15:30:30.904Z: time="2024-01-29T15:30:30Z" level=info msg="Get '-'"
2024-01-29T15:30:31.035Z: time="2024-01-29T15:30:31Z" level=info msg="Get '-'"
2024-01-29T15:30:31.207Z: time="2024-01-29T15:30:31Z" level=info msg="Get '-'"

**Comment by miton18** [comment]
something interresting
```
2024-01-29T15:32:35.608Z: time="2024-01-29T15:32:35Z" level=info msg="Get '-'"
2024-01-29T15:32:35.872Z: time="2024-01-29T15:32:35Z" level=info msg="Get '-'"
[STACKTRACE]
an error occured: [object Event]
[/STACKTRACE]
[ERROR] an error occured: [object Event]
[DEBUG] stream opened! {"appId":"app_fcdbc590-c6d1-41c1-bc22-7b06d0ea76e5","filter":null,"deploymentId":null}
2024-01-29T15:32:35.895Z: time="2024-01-29T15:32:35Z" level=info msg="Get '-'"
2024-01-29T15:32:36.306Z: time="2024-01-29T15:32:36Z" level=info msg="Get '-'"

because
```
logStream
    .on('open', (event) => {
      Logger.debug(`stream opened! ${JSON.stringify({ appId, filter, deploymentId })}`);
    })
    .on('error', (event) => {
      Logger.error(`an error occured: ${event}`);
    })

**Comment by miton18** [comment]
LGTM @hsablonniere

**Review by miton18 — APPROVED** [review]

**Pull request merged**